plugin/pkg/scheduler/factory/factory.go

```diff
@@ -170,12 +171,38 @@ func (factory *ConfigFactory) createMinionLW() *cache.ListWatch {
 
 // pollMinions lists all minions and returns an enumerator for cache.Poller.
 func (factory *ConfigFactory) pollMinions() (cache.Enumerator, error) {
-	list := &api.NodeList{}
-	err := factory.Client.Get().Resource("minions").Do().Into(list)
+	allNodes := &api.NodeList{}
+	err := factory.Client.Get().Resource("minions").Do().Into(allNodes)
 	if err != nil {
 		return nil, err
 	}
-	return &nodeEnumerator{list}, nil
+	nodes := &api.NodeList{
+		TypeMeta: allNodes.TypeMeta,
+		ListMeta: allNodes.ListMeta,
+	}
+	for _, node := range allNodes.Items {
+		if len(node.Status.Conditions) == 0 {
+			// If no condition is set, node health check is disabled (master flag
+			// healthCheckMinions is set to false), we add nodes unconditionally.
+			glog.Infof("No condition set for node %v", node.Name)
+			nodes.Items = append(nodes.Items, node)
+			continue
+		}
+		for _, condition := range node.Status.Conditions {
+			// TODO: Distinguish different conditions and implement better condition check.
+			switch condition.Kind {
+			case api.NodeReachable, api.NodeReady:
```

Comment:
And delete pkg/client/cache/poller*.go since scheduler was only use of that.

On Mon, Jan 12, 2015 at 10:56 AM, Daniel Smith <notifications@github.com>
wrote:

> In plugin/pkg/scheduler/factory/factory.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3389#discussion-diff-22812140>
> :
>
> > +	nodes := &api.NodeList{
> > +		TypeMeta: allNodes.TypeMeta,
> > +		ListMeta: allNodes.ListMeta,
> > +	}
> > +	for _, node := range allNodes.Items {
> > +		if len(node.Status.Conditions) == 0 {
> > +			// If no condition is set, node health check is disabled (master flag
> > +			// healthCheckMinions is set to false), we add nodes unconditionally.
> > +			glog.Infof("No condition set for node %v", node.Name)
> > +			nodes.Items = append(nodes.Items, node)
> > +			continue
> > +		}
> > +		for _, condition := range node.Status.Conditions {
> > +			// TODO: Distinguish different conditions and implement better condition check.
> > +			switch condition.Kind {
> > +			case api.NodeReachable, api.NodeReady:
>
> Please do something to make sure you don't add a node twice...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3389/files#r22812140>
> .
>